src/math/SVI.sol

```diff
@@ -56,9 +69,35 @@ library SVI {
 
     if (w < 0) {
       revert SVI_InvalidParameters();
+    } else if (w > MAX_TOTAL_VAR) {
+      w = MAX_TOTAL_VAR;
     }
 
-    // sqrt((a + b * (sqrt((k - m)^2 + sigma^2) + rho * (k - m)))/tao)
-    return FixedPointMathLib.sqrt(uint(w).divideDecimal(uint(tao)));
+    // sqrt((a + b * (sqrt((k - m)^2 + sigma^2) + rho * (k - m)))/tau)
+    uint vol = FixedPointMathLib.sqrt(uint(w).divideDecimal(uint(tau)));
+    return vol;
+  }
+
+  /**
+   * @dev k = ln(strike / fwd), but being bounded by B: -B < k < B
+   * where B = 3.5 x sqrt(a + b * sig)
+   */
+  function getK(uint strike, int a, uint b, uint sigma, uint forwardPrice) internal pure returns (int k) {
+    // calculate the bounds
+    int volFactor = int(FixedPointMathLib.sqrt((a + b.multiplyDecimal(sigma).toInt256()).toUint256()));
+    int k_bound = volFactor.multiplyDecimal(K_SCALER);
+    int sk = int(strike.divideDecimal(forwardPrice));
+    if (sk == 0) {
+      k = -k_bound;
+    } else {
+      // k = ln (strike / fwd)
+      k = FixedPointMathLib.ln(sk);
+      // make sure -B < k < B
+      if (k > k_bound) {
+        k = k_bound;
+      } else if (k < -k_bound) {
+        k = -k_bound;
```

Comment:
would do explicit returns here (i.e. `return -k_bound;`) to exit early, a bit cleaner imo

Reply:
nice suggestion, i was previously calculating k in the `getVol` function so has to do assigning. we can actually clean it up by quite a lot by returning early in the function